lightning/src/ln/payment_tests.rs

```diff
@@ -266,3 +274,410 @@ fn no_pending_leak_on_initial_send_failure() {
 
 	assert!(!nodes[0].node.has_pending_payments());
 }
+
+#[test]
+fn retry_with_no_persist() {
```

Comment:
Hmm, the new version of this test passes with the code that had the `find_map` bug